lib/net/ldap/connection.rb

```diff
@@ -51,6 +51,15 @@ def open_connection(server)
     hosts.each do |host, port|
       begin
         prepare_socket(server.merge(socket: @socket_class.new(host, port, socket_opts)))
+        if encryption
+          if encryption[:tls_options] &&
+             encryption[:tls_options][:verify_mode] &&
+             encryption[:tls_options][:verify_mode] == OpenSSL::SSL::VERIFY_NONE
+            warn "not verifying SSL hostname of LDAPS server"
+          else
+            @conn.post_connection_check(host)
+          end
+        end
```

Comment:
@jch, I also wanted to put the code into ```.wrap_with_ssl``` initially. However, the current ```.wrap_with_ssl``` function doesn't have access to any variable with the FQDN/hostname used to make the connection in order to pass that to ```post_connection_check```. There was another PR that attempted address this passing around the extra host info by adding an extra host parameter for the ```.wrap_with_ssl``` function. 

See: https://github.com/ruby-ldap/ruby-net-ldap/pull/262#issuecomment-176335135. However, that PR had a few other unresolved questions about how it handled default validation...

One of the reasons I didn't go that way is I note the connection.rb code is flagged for some major refactoring anyhow, and yeah, and as above, you'd be in a better place to make a call where best to inject ```post_connection_check```.